code/modules/mining/minebot.dm

```diff
@@ -154,6 +153,17 @@
 		visible_message("\icon[src] <span class = 'notice'>The [usr] pings, \"[P.name] ready for review\", and happily disgorges a small printout.</span>", range = 2)
 		playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
 
+/mob/living/silicon/robot/drone/mining/process_level_restrictions()
+	//Abort if they should not get blown
+	if (lockcharge || scrambledcodes || emagged)
+		return
+	//Check if they are not on a station level -> abort
+	var/turf/T = get_turf(src)
+	if (!T || isStationLevel(T.z))
```

Comment:
Maybe make it check if drone is leaving it's parent level? Save on initilization it's initial z level if then check if it's origin level is station then check if current level not station and then check if current level is not origin level.

Derelict drones migth explode on spawning

Reply:
That´s the code for the mining bots.
They can only be purchased via the mining vendor on the station -> Hence it is not possible that they explode on spawn.

And for the construction drones that is overkill aswell, as they are created via the `/obj/machinery/drone_fabricator/derelict` which does not exist on station levels

Reply:
As mentioned above it seems that a mining vendor also exists on the scrapheap.
I´ll just remove that one since it would spawn mining drones with the NT Mining drone lawset.